internal/pkg/override/patch.go

```diff
@@ -0,0 +1,318 @@
+			Value: pointer.FinalKey(),
+		})
+		parent.Content = append(parent.Content, &p.Value)
+	case yaml.SequenceNode:
+		if pointer.FinalKey() == "-" {
+			// add to end of sequence
+			parent.Content = append(parent.Content, &p.Value)
+			return nil
+		}
+
+		idx, err := idxOrError(pointer.FinalKey(), len(parent.Content), pointer.Parent())
+		if err != nil {
+			return err
+		}
+
+		// add node at idx
+		parent.Content = append(parent.Content[:idx], append([]*yaml.Node{&p.Value}, parent.Content[idx:]...)...)
+	default:
+		return &errInvalidNodeKind{
+			pointer: pointer.Parent(),
+			kind:    parent.Kind,
+		}
+	}
+
+	return nil
+}
+
+func (p *yamlPatch) applyRemove(root *yaml.Node) error {
+	pointer := p.Pointer()
+	parent, err := getNode(root, pointer.Parent(), nil)
+	if err != nil {
+		return err
+	}
+
+	switch parent.Kind {
+	case yaml.DocumentNode:
+		// make sure we are encoding zero into node
+		p.Value = yaml.Node{}
+		return parent.Encode(p.Value)
+	case yaml.MappingNode:
+		i, err := findInMap(parent, pointer.FinalKey(), pointer.Parent())
+		if err != nil {
+			return err
+		}
+
+		parent.Content = append(parent.Content[:i], parent.Content[i+2:]...)
+	case yaml.SequenceNode:
+		idx, err := idxOrError(pointer.FinalKey(), len(parent.Content)-1, pointer.Parent())
+		if err != nil {
+			return err
+		}
+
+		parent.Content = append(parent.Content[:idx], parent.Content[idx+1:]...)
+	default:
+		return &errInvalidNodeKind{
+			pointer: pointer.Parent(),
+			kind:    parent.Kind,
+		}
+	}
+
+	return nil
+}
+
+func (p *yamlPatch) applyReplace(root *yaml.Node) error {
+	if p.Value.IsZero() {
+		return fmt.Errorf("value required")
+	}
+
+	pointer := p.Pointer()
+	node, err := getNode(root, pointer, nil)
+	if err != nil {
+		return err
+	}
+
+	return node.Encode(p.Value)
+}
+
+type pointer []string
+
+// Parent returns a pointer to the parent of p.
+func (p pointer) Parent() pointer {
+	if len(p) == 0 {
+		return nil
+	}
+	return p[:len(p)-1]
+}
+
+func (p pointer) FinalKey() string {
+	if len(p) == 0 {
+		return ""
+	}
+	return p[len(p)-1]
+}
+
+func (y yamlPatch) Pointer() pointer {
+	split := strings.Split(y.Path, "/")
+	for i := range split {
+		// apply replacements as described https://www.rfc-editor.org/rfc/rfc6901#section-4
+		split[i] = strings.ReplaceAll(split[i], "~1", "/")
+		split[i] = strings.ReplaceAll(split[i], "~0", "~")
+	}
+
+	return split
+}
+
+// findInMap returns the index of the _key_ node in a mapping node's Content.
+// The index of the _value_ node is the returned index+1.
+//
+// If key is not in the map, an error is returned.
+func findInMap(node *yaml.Node, key string, traversed pointer) (int, error) {
+	for i := 0; i < len(node.Content); i += 2 {
+		if node.Content[i].Value == key {
+			return i, nil
+		}
+	}
+
+	return 0, fmt.Errorf("key %q: %q not found in map", strings.Join(traversed, jsonPointerSeparator), key)
+}
+
+func getNode(node *yaml.Node, remaining, traversed pointer) (*yaml.Node, error) {
```

Comment:
maybe consider making the name more meaningful since it is a crucial one. Like `findTargetNode` or sth similar.